src/corecel/Assert.hh

```diff
@@ -180,7 +180,7 @@
     } while (0)
 //! \endcond
 
-#if CELERITAS_DEBUG && CELER_DEVICE_COMPILE
+#if CELERITAS_DEBUG && (CELER_DEVICE_COMPILE || defined(__HIP__))
```

Comment:
Doesn't this use HIP automatically through `CELER_DEVICE_COMPILE`?
```cpp
#if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
#define CELER_DEVICE_COMPILE 1
#else
...
```

Reply:
Ok, I guess this is a munged ramification of the fact that `HIP-Clang` does a single pass, so its not enough that this is a device compile, but that it is potentially using HIP to build host-side code for this translation unit as well, so you are further restricting this to only apply during the device-side compilation?

Reply:
And actually, I just read your overview in more detail where you address this specifically.

Reply:
> clang uses merged parsing. This is similar to split compilation, except all of the host and device code is present and must be semantically-correct in both compilation steps.

So in clang when building host code you can't have device code present that throws, which is different from nvcc which entirely removes device code from the host pass.